packages/cli/generators/datasource/index.js

```diff
@@ -287,6 +288,27 @@ module.exports = class DataSourceGenerator extends ArtifactGenerator {
     // Copy Templates
     this.fs.writeJSON(jsonPath, ds);
     this.copyTemplatedFiles(classTemplatePath, tsPath, this.artifactInfo);
+
+    if (ds.connector === 'memory') {
+      if (!this.artifactInfo.settings.file) {
+        const suggestedFile = './' + this.artifactInfo.name + '.db.json';
+        if (fse.existsSync(suggestedFile)) {
+          this.artifactInfo.settings.file = './' + generateRandomFileName();
+        } else {
+          this.artifactInfo.settings.file = suggestedFile;
+        }
+      }
+      const dbFilePath = path.resolve(this.artifactInfo.settings.file);
+      fse.ensureFileSync(dbFilePath);
+    }
+
+    function generateRandomFileName() {
```

Comment:
I'm wondering instead of just a random file, would it be better to have the name prefixed by the artifact name? Just a thought.

Reply:
Good suggestion.